pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -0,0 +1,254 @@
+// be the integrated authentication
+func (c *webhookAuthenticatorController) ensureOpenShiftAuthConfigured(ctx context.Context,
+	authConfig *configv1.Authentication,
+	recorder events.Recorder,
+) []operatorv1.OperatorCondition {
+	if authConfig.Spec.Type != configv1.AuthenticationTypeIntegratedOAuth && len(authConfig.Spec.Type) != 0 {
+		// not using integrated openshift auth
+		return nil
+	}
+
+	conditions := []operatorv1.OperatorCondition{}
+	oauthAPIsvc, apiSvcConditions := c.getOAuthAPIServerService()
+	if len(apiSvcConditions) > 0 {
+		return append(conditions, apiSvcConditions...)
+	}
+
+	kubeConfigSecret, err := c.ensureKubeConfigSecret(ctx, oauthAPIsvc, recorder)
+	if err != nil {
+		return append(conditions, operatorv1.OperatorCondition{
+			Type:    "AuthenticatorKubeConfigSyncDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "KubeConfigEnsureFailed",
+			Message: err.Error(),
+		})
+	}
+
+	if authConfig.Spec.WebhookTokenAuthenticator == nil {
+		authConfig.Spec.WebhookTokenAuthenticator = &configv1.WebhookTokenAuthenticator{}
+	}
+
+	if authConfig.Spec.WebhookTokenAuthenticator.KubeConfig.Name != kubeConfigSecret.Name {
+		authConfigCopy := authConfig.DeepCopy()
+		authConfigCopy.Spec.WebhookTokenAuthenticator.KubeConfig.Name = kubeConfigSecret.Name
+		_, err := c.authentication.Update(ctx, authConfigCopy, metav1.UpdateOptions{})
+		if err != nil {
+			return append(conditions, operatorv1.OperatorCondition{
+				Type:    "AuthenticatorAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "UpdateFailed",
+				Message: err.Error(),
+			})
+		}
+	}
+
+	return conditions
+}
+
+// getOAuthAPIServerService retrieves the openshift-oauth-apiserver/api service
+func (c *webhookAuthenticatorController) getOAuthAPIServerService() (*corev1.Service, []operatorv1.OperatorCondition) {
+	conditions := []operatorv1.OperatorCondition{}
+	svc, err := c.svcLister.Services("openshift-oauth-apiserver").Get("api")
+	if err != nil {
+		conditions = append(conditions, operatorv1.OperatorCondition{
+			Type:    "AuthenticatorOAuthAPIServiceDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "FailedGet",
+			Message: fmt.Sprintf("failed to retrieve service openshift-oauth-apiserver/oauth-apiserver: %v", err),
+		})
+	}
+	return svc, conditions
+}
+
+// ensureKubeConfigSecret retrieves the token from a secret for the
+// openshift-oauth-apiserver/openshift-authenticator SA. It checks that the
+// "service-ca.crt" and "token" keys are present and creates a
+// openshift-config/webhook-authentication-integrated-oauth secret with a kubeconfig
+// pointing to the oauth-apiserver's tokenvalidation endpoint
+func (c *webhookAuthenticatorController) ensureKubeConfigSecret(ctx context.Context, svc *corev1.Service, recorder events.Recorder) (*corev1.Secret, error) {
+	sa, err := c.saLister.ServiceAccounts("openshift-oauth-apiserver").Get("openshift-authenticator")
```

Comment:
no. This causes cyclical availability issues.  @sttts I think we need a client cert based user.

@stlaz can you manually see if the kubeconfig file used for token authenticators auto-refreshes a referenced client-cert/key pair when specified as a file reference?

Reply:
I tried that with the cert/key file references, they appear to be able to auto-refresh that